### ISSUE TITLE
Type hint on overload implementation should not be considered in compatibility check of method override.

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#3653]
In strict mode, following code reports false error:

```python
class Base:
    def foo(self, x: int) -> int:
        return x

class Derived(Base):
    @overload
    def foo(self, x: int) -> int:
        ...
    @overload
    def foo(self, x: str) -> str:
        ...
    def foo(self, x: int | str) -> int | str:
#       ^^^Method "foo" overrides class "Base" in an incompatible manner...
        return x
```

Seems pyright takes type hint on overload implementation into consideration when checking method override compatibility. This behavior seems wrong as the whole idea of using overload is to avoid annotate a function with single signature.

version: pylance vscode extension v2022.6.30

**Comment by erictraut** [comment]
Pyright's behavior is correct here.

I presume that you have enabled the strict-mode check `reportIncompatibleMethodOverride`. This check is designed to catch cases where you are overriding a method from a base class with an incompatible signature. 

The behavior in this case has nothing to do with overloads. You'll receive the same error if you delete the overload declarations.

**Comment by erictraut** [comment]
Oh, I just realized that I misinterpreted your bug report. You're saying that pyright _should_ take into account the overload signatures when determining whether the override is compatible with the base class. OK, I think that's reasonable.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.259, which I just published. It will also be included in the next release of pylance.
